.github/workflows/cargo-tests.reusable.yaml

```
@@ -153,93 +163,128 @@ jobs:
+  # macOS: BUILD on a fast Linux runner (cross-compile to aarch64-apple-darwin via
+  # cargo-zigbuild), then RUN the prebuilt binaries on a Mac (no compile). The
+  # cross-build is faster + cheaper than native Mac; ring-crypto drops aws-lc-sys
+  # (which can't cross-build to arm64-darwin), and the Apple SDK is supplied from
+  # R2 (packaged from our Xcode; not redistributed). build-caching/04 §7b.
+  # NOTE: needs R2 secrets (fork PRs without them can't pull the SDK — add a
+  # native-Mac fallback for forks if needed). x64 host chosen per Exp 18.
+  cargo-test-macos-archive:
+    name: "cargo test (macos) — archive on linux"
     if: ${{ github.event_name != 'merge_group' }}
-    runs-on: blacksmith-6vcpu-macos-latest
+    runs-on: blacksmith-8vcpu-ubuntu-2404
```

actionlint (1.7.12)

[error] label "blacksmith-8vcpu-ubuntu-2404" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows", "blacksmith-16vcpu-ubuntu-2404", "blacksmith-8vcpu-ubuntu-2204-arm", "blacksmith-4vcpu-ubuntu-2404", "blacksmith-4vcpu-ubuntu-2204", "blacksmith-4vcpu-ubuntu-2204-arm", "blacksmith-4vcpu-windows-2025". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)